src/systemctl/systemctl.c

```diff
@@ -2098,7 +2097,7 @@ static int set_default(int argc, char *argv[], void *userdata) {
                 if (r < 0)
                         return log_error_errno(r, "Failed to set default target: %s", bus_error_message(&error, r));
 
-                r = bus_deserialize_and_dump_unit_file_changes(reply, arg_quiet, NULL, NULL);
+                r = bus_deserialize_and_dump_unit_file_changes(reply, arg_quiet, &changes, &n_changes);
```

Comment:
@eworm-de please merge these commits, so that there's not intermediary state of the repo with the mem leak in place...